src/WebUI/ImageDetails/ImageDetails.tsx

```diff
@@ -281,5 +297,84 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
         return "";
     }
 
+    private _getImageLabels(layerInfo: IImageLayer[]): string[] {
+        let labels: string[] = [];
+        for (const layer of layerInfo) {
+            const directive = layer.directive;
+            if (directive.toLowerCase() === "label") {
+                labels.push(layer.arguments);
+            }
+        }
+
+        return labels;
+    }
+
+    private _constructPipelineRunUrl(labels: string[], runId: number): string {
+        // These strings are the ones that are added as labels during docker build task
+        const teamFoundationCollectionUriMatch = "system.teamfoundationcollectionuri=";
+        const teamProjectMatch = "system.teamproject=";
+        const buildNumberMatch = "build.buildnumber=";
+        const releaseWebUrlMatch = "release.releaseweburl=";
+        let teamFoundationCollectionUri = "";
+        let teamProject = "";
+        let buildNumber = "";
+        let releaseWebUrl = "";
+
+        for (const label of labels) {
+            releaseWebUrl = this._getMatchingUrlString(label, releaseWebUrlMatch);
+        }
+
+        // For release pipeline, the releaseWebUrl is added as a tag during docker build
+        if (!!releaseWebUrl) {
+            return releaseWebUrl;
+        }
+
+        // For build pipeline we construct url from bits found by searching each of the labels
+        for (const label of labels) {
+            teamFoundationCollectionUri = this._getMatchingUrlString(label, teamFoundationCollectionUriMatch);
+            if (!!teamFoundationCollectionUri) {
+                break;
+            }
+        }
+
+        if (!teamFoundationCollectionUri) {
+            return "";
+        }
+
+        for (const label of labels) {
+            teamProject = this._getMatchingUrlString(label, teamProjectMatch);
+            if (!!teamProject) {
+                break;
+            }
+        }
+
+        if (!teamProject) {
+            return "";
+        }
+
+        for (const label of labels) {
+            buildNumber = this._getMatchingUrlString(label, buildNumberMatch);
+            if (!!buildNumber) {
+                break;
+            }
+        }
+
+        if (!buildNumber) {
+            return "";
+        }
+
+        return format("{0}{1}/_build?buildId={2}", teamFoundationCollectionUri, teamProject, runId.toString());
```

Comment:
is this the rightway o build the URL, should not we rely on the image details itself, this could change in future. or this will not change?

Reply:
We are not sending this data in image details Grafeas format currently, as it is specific to AzureDevOps. We extract this from labels set using docker task. Maybe we should add the build url as a label in Docker task itself, just like we add releaseWebUrl. But currently it is not there